src/commands/updateContainer/updateContainer.ts

```diff
@@ -0,0 +1,70 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { type Revision } from "@azure/arm-appcontainers";
+import { AzureWizard, createSubscriptionContext, type IActionContext, type ISubscriptionContext } from "@microsoft/vscode-azext-utils";
+import { type ContainerAppModel } from "../../tree/ContainerAppItem";
+import { type ContainerItem } from "../../tree/containers/ContainerItem";
+import { ContainersItem } from "../../tree/containers/ContainersItem";
+import { createActivityContext } from "../../utils/activityUtils";
+import { getManagedEnvironmentFromContainerApp } from "../../utils/getResourceUtils";
+import { getVerifyProvidersStep } from "../../utils/getVerifyProvidersStep";
+import { localize } from "../../utils/localize";
+import { pickContainer } from "../../utils/pickItem/pickContainer";
+import { getParentResourceFromItem, isTemplateItemEditable, throwTemplateItemNotEditable } from "../../utils/revisionDraftUtils";
+import { ImageSourceListStep } from "../image/imageSource/ImageSourceListStep";
+import { RevisionDraftDeployPromptStep } from "../revisionDraft/RevisionDraftDeployPromptStep";
+import { type ContainerUpdateContext } from "./ContainerUpdateContext";
+import { ContainerUpdateDraftStep } from "./ContainerUpdateDraftStep";
+import { RegistryAndSecretsUpdateStep } from "./RegistryAndSecretsUpdateStep";
+
+// Updates both the 'image' and 'environmentVariables' portion of the container profile
+export async function updateContainer(context: IActionContext, node?: ContainersItem | ContainerItem): Promise<void> {
+    const item: ContainerItem | ContainersItem = node ?? await pickContainer(context, { autoSelectDraft: true });
+    const { containerApp, subscription } = item;
+
+    if (!isTemplateItemEditable(item)) {
+        throwTemplateItemNotEditable(item);
+    }
+
+    const subscriptionContext: ISubscriptionContext = createSubscriptionContext(subscription);
+    const parentResource: ContainerAppModel | Revision = getParentResourceFromItem(item);
+
+    let containersIdx: number;
+    if (ContainersItem.isContainersItem(item)) {
+        // The 'updateContainer' command should only show up on a 'ContainersItem' when it only has one container, else the command would show up on the 'ContainerItem'
```

Comment:
This is a nit, but if it's an easy change you might want to consider it.

I think that it might be better to always show the command on the Container**s**Item. Can we do that and when there are multiple containers just ask the user to pick one? You already have the pick container logic above, you might be able to reuse that.

I believe that users don't like when commands are missing, we want to be consistent so they can do things in the same way each time

Reply:
Yeah that's a good point, that's also how we have it implemented for picking revisions so this would help keep that consistent.  I will make a change in a follow-up PR to add this